src/ethereum/EthereumEngine.ts

```diff
@@ -775,6 +786,41 @@ export class EthereumEngine extends CurrencyEngine<
     await this.startEngine()
   }
 
+  async syncNetwork(opts: EdgeEngineSyncNetworkOptions): Promise<number> {
```

Comment:
We have added this new logic, but we never got rid of `this.setupAdapterSubscriptions()` over in EthereumNetwork, so we are now subscribed twice? I would have expected more deleted code.